pageserver/src/tenant/timeline.rs

```diff
@@ -3538,12 +3548,33 @@ impl Timeline {
                                     ?other,
                                     "replacing downloaded layer into layermap failed because another layer was found instead of expected"
                                 );
+                                true
                             }
                             Err(e) => {
                                 // this is a precondition failure, the layer filename derived
                                 // attributes didn't match up, which doesn't seem likely.
-                                error!("replacing downloaded layer into layermap failed: {e:#?}")
+                                error!("replacing downloaded layer into layermap failed: {e:#?}");
+                                true
                             }
+                        };
+
```

Comment:
A failpoint here AND a logical size query or synthetic size query allow use to demonstrate that it doesn't busy loop at least?

Reply:
Well, we cannot do a failpoint here *after* successful replacement, we could just replace the replacement with a failpoint to mock it out.

Reply:
Done in [5c54eda](https://github.com/neondatabase/neon/pull/3613/commits/5c54eda8aa06d8280544b29f932a2b2ebfd55fc5)